File: app/src/main/java/com/example/water/cproject/Genuino/Gyroscope.java
package com.example.water.cproject.Genuino;

/**
 * Created by water on 2017-04-18.
 */

@SuppressWarnings("DefaultFileTemplate")
public class Gyroscope {
    private float x, y, z;

    public Gyroscope() {
        updateData(0,0,0);
    }

    public void updateData(float x, float y, float z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
    public Gyroscope getData() {
        return this;
    }
    public float getX() {
        return x;
    }
    public float getY() {
        return y;
    }
    public float getZ() {
        return z;
    }
}

File: app/src/main/java/com/example/water/cproject/genuino/Gyroscope.kt
package com.example.water.cproject.genuino

/**
 * Created by water on 2017-04-18.
 */

class Gyroscope {
    var x: Float = 0.toFloat()
        private set
    var y: Float = 0.toFloat()
        private set
    var z: Float = 0.toFloat()
        private set
    val data: Gyroscope
        get() = this

    init {
        updateData(0f, 0f, 0f)
    }

    fun updateData(x: Float, y: Float, z: Float) {
        this.x = x
        this.y = y
        this.z = z
    }
}
